### DISCUSSION TITLE
how the scrambled JWT is descrambled.

### DISCUSSION THREAD

**Discussion by buddhadonthavemoney** [basic-bgnr/NepseUnofficialApi#22] [Q&A] [answered]
I also had created a CLI tool and nepse library about three years ago, [NEPSENSE](https://github.com/buddhadonthavemoney/NEPSENSE). Before the **prove** endpoint was introduced, it worked. But due to this authorization scheme, I wasn't able to make it work. I didn't understand how the scrambled JWT token was unscrambled and stopped trying and archived the repository. Now I am using your token utils and my previous project is working. Can you tell me what was your approach to solving this problem?. It could be really helpful. Thank you!!

**Comment by basic-bgnr** [comment, marked as answer]
@buddhadonthavemoney. I had to go through the client-side js source to find it out. Network request scanning was just not sufficient. To make the matter more cumbersome, Nepse used compiled wasm to generate the token indices. So it required some fiddling around.

**Reply by buddhadonthavemoney** [reply]
Wow, bro, that's a lot of work and kudos to your problem-solving skill. The JS code must have been a mess. I also tried looking at it, but my head hurt. Did you decompile the Wasm code as well, or studied the front end calling the Wasm functions? 

But thanks to your handwork now my code works lol. You won't mind if I use the token utils in my library, will you?

**Reply by basic-bgnr** [reply]
>  You won't mind if I use the token utils in my library, will you?

@buddhadonthavemoney not at all.  Go for it.
